src/app/components/category-products/category-products.component.ts

```diff
@@ -8,4 +20,110 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class CategoryProductsComponent {
+
+  private _sortings$: Observable<CategoryProductsSortQueryModel[]> = of([
+    { id: ProductSorting.featureValueDesc, name: 'Featured' },
+    { id: ProductSorting.priceAsc, name: 'Price: Low to High' },
+    { id: ProductSorting.priceDesc, name: 'Price: High to Low' },
+    { id: ProductSorting.ratingValueDesc, name: 'Avg. Rating' }
+  ]);
+
+  private _filters$: Observable<CategoryProductsFiltersQueryModel> = combineLatest([
+    this._activatedRoute.params,
+    this._activatedRoute.queryParams,
+    this._sortings$
+  ]).pipe(
+    map(
+      ([params, queryParams, sortings]: [Params, Params, CategoryProductsSortQueryModel[]]) => ({
+        categoryId: params['categoryId'],
+        sort: queryParams['sort'] !== undefined ? +queryParams['sort'] : ProductSorting.featureValueDesc,
+        sortings
+    }))
+  );
+
+  readonly model$: Observable<CategoryProductsQueryModel> = combineLatest([
+    this._filters$,
+    this._categoryService.getAll(),
+    this._productService.getAll()
+  ]).pipe(
+    map(
+      (
+        [filters, categories, products]: [CategoryProductsFiltersQueryModel, CategoryModel[], ProductModel[]]
+      ) => this._mapQueryModel(filters, categories, products)
+    )
+  );
+
+  constructor(private _categoryService: CategoryService, private _activatedRoute: ActivatedRoute, private _productService: ProductService, private _router: Router) {
+  }
+
+  public onSortingChange(target: EventTarget | null): void {
+    if (target === null) return;
+
+    const value = (target as HTMLSelectElement).value;
+
+    this._router.navigate(
+      [],
+      {
+        relativeTo: this._activatedRoute,
+        queryParams: {
+          sort: value
+        },
+        queryParamsHandling: 'merge'
+      }
+    )
+  }
+
+  private _filterProducts(filters: CategoryProductsFiltersQueryModel, products: ProductModel[]): ProductModel[] {
+    const sortFunctionsMap = new Map([
```

Comment:
ok cool, you used Map instead of ifs :) but in fact you can do it better with more advanced query model....

Reply:
Technically you want to resolve based on enum value: which property to sort on and in which direction - once you have that, you can use single sort function